### PR TITLE
RavenDB-15089 Add studio configuration to collapse document

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/common/settings/globalSettings.ts [hunk 1]
```diff
@@ -10,8 +10,9 @@ class globalSettings extends abstractSettings {
     private readonly remoteSettingsLoader: () => JQueryPromise<Raven.Client.ServerWide.Operations.Configuration.ServerWideStudioConfiguration>;
     private readonly remoteSettingsSaver: (settings: Raven.Client.ServerWide.Operations.Configuration.ServerWideStudioConfiguration) => JQueryPromise<void>;
 
-    replicationFactor = new simpleStudioSetting<number>(
-        "remote", null, x => this.saveSetting(x));
+    replicationFactor = new simpleStudioSetting<number>("remote", null, x => this.saveSetting(x));
+    
+    collapseDocsWhenOpening = new simpleStudioSetting<boolean>("local", true, x => this.saveSetting(x));
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/settings/globalSettings.ts]: true or false by default? I think we should match current behaviuor

File: src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts [hunk 2]
```diff
@@ -152,6 +152,10 @@ class editDocument extends viewModelBase {
     canViewRevisions: KnockoutComputed<boolean>;
     canViewRelated: KnockoutComputed<boolean>;
     canViewCSharpClass: KnockoutComputed<boolean>;
+
+    collapseDocsWhenOpening = ko.observable<boolean>(false);
+    isDocumentCollapsed = ko.observable<boolean>(false);
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts:156]: is it updated when you manually toggle folds? maybe we should avoid having this field?

File: src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts [hunk 3]
```diff
@@ -213,6 +217,25 @@ class editDocument extends viewModelBase {
         (ace as any).config.loadModule("ace/mode/raven_document_newline_friendly");
 
         this.connectedDocuments.compositionComplete();
+        
+        studioSettings.default.globalSettings()
+            .done((settings: globalSettings) => {
+                if (settings.collapseDocsWhenOpening.getValue()) {
+                    const collapse = settings.collapseDocsWhenOpening.getValue();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts]: value is true  see line above, right?

File: src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts [hunk 4]
```diff
@@ -624,13 +647,13 @@ class editDocument extends viewModelBase {
         }
     }
 
-    toggleAutoCollapse() {
-        eventsCollector.default.reportEvent("document", "toggle-auto-collapse");
-        this.autoCollapseMode.toggle();
-        if (this.autoCollapseMode()) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts:630]: we have auto collapse mode field and isdocument collapsed. can we have single field to represent this state?

File: src/Raven.Studio/typescript/viewmodels/database/settings/studioConfiguration.ts [hunk 5]
```diff
@@ -24,6 +26,11 @@ class studioConfiguration extends viewModelBase {
             });
     }
 
+    compositionComplete() {
+        super.compositionComplete();
+        $('.studio-configuration [data-toggle="tooltip"]').tooltip();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/studioConfiguration.ts]: do we need that? obsolete code?

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/settings/studioConfiguration.ts]: removed

File: src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts]: done

File: src/Raven.Studio/typescript/common/settings/globalSettings.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/common/settings/globalSettings.ts]: done, changed to false

File: src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts:630]: I will remove ```autoCollapseMode```
It is not used with the new implementation

File: src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts:156]: Please look at the latest commit implementation
I fixed bugs for when _cloning_ & when _exiting compare revisions_ state
Seems that field is needed

**Review by ml054 — APPROVED** [review]

**Pull request merged**